### ISSUE TITLE
OpenSearch Dashboards Distribution: Publish to NPM?

### ISSUE THREAD

**Issue by tmarkley** [opensearch-project/OpenSearch-Dashboards#542]
We want to have an open discussion regarding the plan for publishing Dashboards artifacts.

NPM seems like the correct destination, but here are some questions that we need to consider:
* How are plugins currently building against Dashboards?
* How do other third-party developers need to consume Dashboards?

**Comment by dblock** [comment]
What are the alternatives to npm?

**Comment by tmarkley** [comment]
@seraphjiang mentioned previously that we could consider RPM, Debian, and/or Homebrew.

**Comment by dblock** [comment]
> @seraphjiang mentioned previously that we could consider RPM, Debian, and/or Homebrew.

That works for distributed artifacts, but not for node versioned dependencies.

**Comment by camerski** [comment]
NPM seems to be the de-facto standard for node versioned dependencies. Is there a reason to not use NPM, at least as a baseline? That wouldn't block adding other distribution channels in future.

**Comment by tmarkley** [comment]
> That works for distributed artifacts, but not for node versioned dependencies.

> NPM seems to be the de-facto standard for node versioned dependencies. Is there a reason to not use NPM, at least as a baseline? That wouldn't block adding other distribution channels in future.

Agreed on both accounts. I don't think there's a reason not to use NPM considering it seems like there are plenty of use cases where customers would benefit. It's not clear to me why the repo was never published to NPM before.

**Comment by deknos** [comment]
Please provide Debian Packages, perhaps even talk to the debian guys, if you are allowed to push it into the debian repositories

You could also have a look at pkgsrc (netbsd packging, also usable on linux and on Free/Open/NetBSD) and nixos (very good at reproducible builds)

**Comment by elfisher** [comment]
Would each OpenSearch Dashboards plugin artifact be published as its own node package in NPM?
